src/internal/connector/exchange/exchange_data_collection.go

```diff
@@ -84,15 +92,32 @@ func NewCollection(
 		jobs:           make([]string, 0),
 		service:        service,
 		statusUpdater:  statusUpdater,
-		fullPath:       fullPath,
-		prevPath:       prevPath,
+		fullPath:       curr,
+		prevPath:       prev,
 		collectionType: collectionType,
 		ctrl:           ctrlOpts,
+		state:          stateOf(prev, curr),
 	}
 
 	return collection
 }
 
+func stateOf(prev, curr path.Path) data.CollectionState {
+	if curr == nil || len(curr.String()) == 0 {
+		return data.DeletedState
+	}
+
+	if prev == nil || len(prev.String()) == 0 {
+		return data.NewState
+	}
+
+	if !common.AreSameSlice(curr.Folders(), prev.Folders()) {
+		return data.MovedState
+	}
```

Comment:
these are both paths, why not just compare strings? I think `curr.Folder() != prev.Folder()` or `curr.String() != prev.String()` would suffice using only basic comparisons defined in go. This would remove the dependency on slice comparison code as well

Reply:
Oh... you know, because I'm daft.

Reply:
eh, it's easy to get caught up in the moment. I've definitely done it before. I think the common one for me is when I'm using C++ to go through a vector and delete all the object instances and then get null pointer errors cause I forgot to reset the vector 🙃